### PR TITLE
chore(suite): update used address table to use Table component

### LINKED ISSUE #14706: Add and edit label button too long in receive section
**Describe the bug**
Rename label button too long in receive section

**Steps to reproduce:**
1. Enable labeling
2. Navigate to receive section 
3. Hover over address
4. Observe button lenght

**Info:**
 - Suite version: desktop 24.10.1 (3017e5864a176715b1023945397ef49e8aa24d5e)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/24.10.1 Chrome/126.0.6478.234 Electron/31.6.0 Safari/537.36
 - OS: MacIntel
 - Screen: 1512x982
 - Device: Trezor T2T1 2.8.4 regular (revision 454b8140ecb1575a1adafb8e57051e345a32255e)
 - Transport: BridgeTransport 3.0.0-bundled.24.10.1

**Screenshots:**

_**Receive address section**_
Screenshot 2024-10-03 at 12 41 33
image

_**TX history**_
Screenshot 2024-10-03 at 12 41 25

### PR DESCRIPTION
**Pull request by seibei-iguchi**
### Before:
Screenshot 2024-10-08 at 10 44 56

### After:
Screenshot 2024-10-08 at 10 43 06

### CODE REVIEW

**Review by Hermez-cz — APPROVED** [review]

File: packages/components/src/components/Table/TableRow.tsx [hunk 1]
```diff
@@ -68,21 +68,31 @@ export const Row = styled.tr<{
 export interface TableRowProps {
     children: ReactNode;
     isCollapsed?: boolean;
+    highlightRowOnHover?: boolean;
```
Comment (jvaclavik) [inline packages/components/src/components/Table/TableRow.tsx]: What about `isHighlightedOnHover`?

File: packages/components/src/components/Table/TableRow.tsx [hunk 2]
```diff
@@ -68,21 +68,31 @@ export const Row = styled.tr<{
...
     const isHeader = useTableHeader();
-    const { highlightRowOnHover } = useTable();
+    const { highlightRowOnHover: highlightRowOnHoverGlobal } = useTable();
 
     return (
         <Row
             $elevation={elevation}
             $isCollapsed={isCollapsed}
-            $highlight={highlightRowOnHover}
+            $highlight={highlightRowOnHover ?? highlightRowOnHoverGlobal}
```
Comment (jvaclavik) [inline packages/components/src/components/Table/TableRow.tsx]: What about `$isHighlighted`?

It looks a bit inconsistent when you have `$isHeader` and `$isCollapsed` with `is` here

File: packages/suite/src/views/wallet/receive/components/UsedAddresses.tsx [hunk 3]
```diff
@@ -226,37 +162,42 @@ export const UsedAddresses = ({
...
+                            <Table.Cell />
+                            <Table.Cell align="right">
+                                <Translation id="RECEIVE_TABLE_RECEIVED" />
+                            </Table.Cell>
+                        </Table.Row>
+                    </Table.Header>
+                    <Table.Body>
+                        {list.slice(0, limit).map((addr, index) => (
+                            <Item
+                                index={index}
```
Comment (jvaclavik) [inline packages/suite/src/views/wallet/receive/components/UsedAddresses.tsx:182]: Can we replace `index` with `addr.address`?

**Review by jvaclavik — APPROVED** [review]
Just few small comments, otherwise approving

File: packages/suite/src/views/wallet/receive/components/UsedAddresses.tsx (hunk 3 above)
Comment (seibei-iguchi) [reply packages/suite/src/views/wallet/receive/components/UsedAddresses.tsx:182]: What's the reason? It's used for test ids and I would have to update E2E tests?

Comment (jvaclavik) [reply packages/suite/src/views/wallet/receive/components/UsedAddresses.tsx:182]: OK, keep it

**Pull request merged**